src/components/billingRecord/IFXCalculateBillingMonth.vue

```diff
@@ -0,0 +1,271 @@
+<script>
+import { mapActions } from 'vuex'
+import IFXItemDataTable from '@/components/item/IFXItemDataTable'
+
+export default {
+  name: 'IFXCalculateBillingMonth',
+  props: {
+    month: {
+      type: String,
+      required: false,
+    },
+    year: {
+      type: String,
+      required: false
+    },
+    facilityId: {
+      type: String,
+      required: false
+    },
+    organization: {
+      type: String,
+      required: false,
+      default: null
+    }
+  },
+  components: {
+    IFXItemDataTable,
+  },
+  data() {
+    return {
+      dateMenu: false,
+      selectedDate: null,
+      localMonth: null,
+      localYear: null,
+      localOrganization: null,
+      facility: null,
+      facilities: [],
+      interval: null,
+      usages: [],
+      isLoading: true,
+      apiName: null,
+      selected: [],
+      recalculate: false,
+    }
+  },
+  computed: {
+    headers() {
+      const headers = [
+        { text: 'ID', value: 'id', sortable: true },
+        { text: 'User', value: 'productUser', sortable: true, namedSlot: true },
+        { text: 'Year', value: 'year', slot: true, sortable: true },
+        { text: 'Month', value: 'month', slot: true, sortable: true },
+        { text: 'Organization', value: 'organization', namedSlot: true, sortable: true },
+        { text: 'Description', value: 'description', slot: true },
+        { text: 'Processing', value: 'processing', sortable: true, namedSlot: true },
+      ]
+      return headers.filter((h) => !h.hide || !this.$vuetify.breakpoint[h.hide])
+    },
+    calculateBillingMonthToolTip() {
+      if (!this.canCalculate()) {
+        return 'To create new billing for usages that have billing records, check Recalculate'
+      }
+      if (this.recalculate) {
+        return 'Recalculate billing records'
+      }
+      return 'Calculate billing records'
+    }
+  },
+  methods: {
+    ...mapActions(['showMessage']),
+    canCalculate() {
+      if (this.usages && this.usages.some((usage) => this.hasFinalBillingRecord(usage))) {
```

Comment:
This should produce the same result but might not be as clear. Your call.
```suggestion
      if (this.usages?.some((usage) => this.hasFinalBillingRecord(usage))) {
```